include/mbedtls/aes.h

```diff
@@ -57,68 +67,89 @@ extern "C" {
 #endif
 
 /**
- * \brief          AES context structure
+ * \brief The AES context-type definition.
  *
- * \note           buf is able to hold 32 extra bytes, which can be used:
- *                 - for alignment purposes if VIA padlock is used, and/or
- *                 - to simplify key expansion in the 256-bit case by
- *                 generating an extra round key
+ * \note           This buffer can hold 32 extra Bytes, which can be used for
```

Comment:
Blocker: Reading `This buffer` after being introduced to `The AES context-type definition` one associates `buffer` with the AES context, but instead a particular field in the specific structure is meant. The documentation should make that clear. Suggestion: Why not moving the entire comment into the structure field documentation of the `buf` field?